orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationClient.java

```diff
@@ -245,42 +197,20 @@ CompletionPostResponseSynchronous executeRequest(@Nonnull final String request)
   public Stream<OrchestrationChatCompletionDelta> streamChatCompletionDeltas(
       @Nonnull final CompletionPostRequest request) throws OrchestrationClientException {
     request.getOrchestrationConfig().setStream(true);
-    return executeStream("/completion", request, OrchestrationChatCompletionDelta.class);
-  }
-
-  @Nonnull
-  private <D extends StreamedDelta> Stream<D> executeStream(
-      @Nonnull final String path,
-      @Nonnull final Object payload,
-      @Nonnull final Class<D> deltaType) {
-    final var request = new HttpPost(path);
-    serializeAndSetHttpEntity(request, payload);
-    return streamRequest(request, deltaType);
-  }
-
-  private static void serializeAndSetHttpEntity(
-      @Nonnull final BasicClassicHttpRequest request, @Nonnull final Object payload) {
-    try {
-      final var json = JACKSON.writeValueAsString(payload);
-      request.setEntity(new StringEntity(json, ContentType.APPLICATION_JSON));
-    } catch (final JsonProcessingException e) {
-      throw new OrchestrationClientException("Failed to serialize request parameters", e);
-    }
+    return executor.stream(request);
   }
 
+  /**
+   * Generate embeddings for the given request.
+   *
+   * @param request the request containing the input text and other parameters.
+   * @return the response containing the embeddings.
+   * @throws OrchestrationClientException if the request fails
+   * @since 1.9.0
+   */
   @Nonnull
-  private <D extends StreamedDelta> Stream<D> streamRequest(
-      final BasicClassicHttpRequest request, @Nonnull final Class<D> deltaType) {
-    try {
-      val destination = destinationSupplier.get();
-      log.debug("Using destination {} to connect to orchestration service", destination);
-      val client = ApacheHttpClient5Accessor.getHttpClient(destination);
-      return new ClientStreamingHandler<>(
-              deltaType, OrchestrationError.class, OrchestrationClientException::new)
-          .objectMapper(JACKSON)
-          .handleStreamingResponse(client.executeOpen(null, request, null));
-    } catch (final IOException e) {
-      throw new OrchestrationClientException("Request to the Orchestration service failed", e);
-    }
+  public EmbeddingsPostResponse embed(@Nonnull final EmbeddingsPostRequest request)
+      throws OrchestrationClientException {
+    return executor.execute("/v2/embeddings", request, EmbeddingsPostResponse.class);
```

Comment:
(Question)

Is it `/v2/embeddings` and `/completion`? Looks counter-intuitive to me.

Reply:
We checked it, it looks legit. We'll approach Orchestration colleagues on that later.